src/lib/gpm.c

```diff
@@ -0,0 +1,52 @@
+#include "internal.h"
+#ifdef USE_GPM
+#undef buttons // defined by both term.h and gpm.h, ugh
+#include <lib/gpm.h>
+#include <gpm.h>
+
+static Gpm_Connect gpmconn;    // gpm server handle
+
+int gpm_connect(tinfo* ti){
+  (void)ti;
+  gpm_zerobased = 1;
+  gpmconn.eventMask = ~0;
+  gpmconn.defaultMask = 0;
+  gpmconn.minMod = 0;
+  gpmconn.maxMod = ~0;
```

Comment:
I believe this will prevent use of {keyboard modifier}+drag for selecting text on the console.

I chose to leave maxMod = 0 so that one can still Shift-Drag for GPM console copy/paste.

Reply:
cheerfully ripping off your init constants, thanks!

Reply:
changed `maxMod`. btw you've got:

```c
                gpm_connection.eventMask = GPM_MOVE | GPM_DRAG;
                gpm_connection.eventMask |= GPM_DOWN | GPM_UP;
```

which seems to effectively be `~0` with more steps.

Reply:
for the readability check, i need to chuck the fd into my primary input `poll()`; that's why i'm returning either -1 or the fd from `gpm_connect()`. one thing i don't get is why `Gpm_Getc()` wants a `FILE*`. i'm going to be using `Gpm_GetEvent()` anyway, but it weirds me out (the latter does not require a `FILE*`). anyway.

Reply:
https://github.com/dankamongmen/notcurses/commit/bb709acd58df5465e71399003e08050f79fbfe3b

Reply:
https://github.com/dankamongmen/notcurses/commit/e1226783a79499566bf54f6dd51d55873a3759b7 i mean

Reply:
> changed `maxMod`. btw you've got:
> 
> ```c
>                 gpm_connection.eventMask = GPM_MOVE | GPM_DRAG;
>                 gpm_connection.eventMask |= GPM_DOWN | GPM_UP;
> ```
> 
> which seems to effectively be `~0` with more steps.

Yup!  You'll notice I tend to go for extreme verbosity, even with things like 'if (blah == false)'.  It's because my memory is pretty bad. ;-)